{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "622a96e8",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'isrgrootx1.pem'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[1], line 9\u001b[0m\n\u001b[0;32m      5\u001b[0m mongodb_url_key \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmongodb+srv://clarityai:N0lTZ4zedDxGbn9N@cluster0.qketnjc.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0&tlsCAFile=isrgrootx1.pem\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m      8\u001b[0m \u001b[38;5;66;03m# Create a new client and connect to the server\u001b[39;00m\n\u001b[1;32m----> 9\u001b[0m client \u001b[38;5;241m=\u001b[39m MongoClient(mongodb_url_key)\n\u001b[0;32m     11\u001b[0m \u001b[38;5;66;03m# Send a ping to confirm a successful connection\u001b[39;00m\n\u001b[0;32m     12\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pymongo\\mongo_client.py:771\u001b[0m, in \u001b[0;36mMongoClient.__init__\u001b[1;34m(self, host, port, document_class, tz_aware, connect, type_registry, **kwargs)\u001b[0m\n\u001b[0;32m    767\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m    768\u001b[0m     timeout \u001b[38;5;241m=\u001b[39m common\u001b[38;5;241m.\u001b[39mvalidate_timeout_or_none_or_zero(\n\u001b[0;32m    769\u001b[0m         keyword_opts\u001b[38;5;241m.\u001b[39mcased_key(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mconnecttimeoutms\u001b[39m\u001b[38;5;124m\"\u001b[39m), timeout\n\u001b[0;32m    770\u001b[0m     )\n\u001b[1;32m--> 771\u001b[0m res \u001b[38;5;241m=\u001b[39m uri_parser\u001b[38;5;241m.\u001b[39mparse_uri(\n\u001b[0;32m    772\u001b[0m     entity,\n\u001b[0;32m    773\u001b[0m     port,\n\u001b[0;32m    774\u001b[0m     validate\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[0;32m    775\u001b[0m     warn\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[0;32m    776\u001b[0m     normalize\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    777\u001b[0m     connect_timeout\u001b[38;5;241m=\u001b[39mtimeout,\n\u001b[0;32m    778\u001b[0m     srv_service_name\u001b[38;5;241m=\u001b[39msrv_service_name,\n\u001b[0;32m    779\u001b[0m     srv_max_hosts\u001b[38;5;241m=\u001b[39msrv_max_hosts,\n\u001b[0;32m    780\u001b[0m )\n\u001b[0;32m    781\u001b[0m seeds\u001b[38;5;241m.\u001b[39mupdate(res[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnodelist\u001b[39m\u001b[38;5;124m\"\u001b[39m])\n\u001b[0;32m    782\u001b[0m username \u001b[38;5;241m=\u001b[39m res[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124musername\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;129;01mor\u001b[39;00m username\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pymongo\\uri_parser.py:529\u001b[0m, in \u001b[0;36mparse_uri\u001b[1;34m(uri, default_port, validate, warn, normalize, connect_timeout, srv_service_name, srv_max_hosts)\u001b[0m\n\u001b[0;32m    526\u001b[0m     dbase \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m    528\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m opts:\n\u001b[1;32m--> 529\u001b[0m     options\u001b[38;5;241m.\u001b[39mupdate(split_options(opts, validate, warn, normalize))\n\u001b[0;32m    530\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m srv_service_name \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m    531\u001b[0m     srv_service_name \u001b[38;5;241m=\u001b[39m options\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msrvServiceName\u001b[39m\u001b[38;5;124m\"\u001b[39m, SRV_SERVICE_NAME)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pymongo\\uri_parser.py:363\u001b[0m, in \u001b[0;36msplit_options\u001b[1;34m(opts, validate, warn, normalize)\u001b[0m\n\u001b[0;32m    360\u001b[0m     options \u001b[38;5;241m=\u001b[39m _normalize_options(options)\n\u001b[0;32m    362\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m validate:\n\u001b[1;32m--> 363\u001b[0m     options \u001b[38;5;241m=\u001b[39m cast(_CaseInsensitiveDictionary, validate_options(options, warn))\n\u001b[0;32m    364\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m options\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mauthsource\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m    365\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m InvalidURI(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mthe authSource database cannot be an empty string\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pymongo\\uri_parser.py:321\u001b[0m, in \u001b[0;36mvalidate_options\u001b[1;34m(opts, warn)\u001b[0m\n\u001b[0;32m    308\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mvalidate_options\u001b[39m(opts: Mapping[\u001b[38;5;28mstr\u001b[39m, Any], warn: \u001b[38;5;28mbool\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m MutableMapping[\u001b[38;5;28mstr\u001b[39m, Any]:\n\u001b[0;32m    309\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Validates and normalizes options passed in a MongoDB URI.\u001b[39;00m\n\u001b[0;32m    310\u001b[0m \n\u001b[0;32m    311\u001b[0m \u001b[38;5;124;03m    Returns a new dictionary of validated and normalized options. If warn is\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    319\u001b[0m \u001b[38;5;124;03m          cause errors.\u001b[39;00m\n\u001b[0;32m    320\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m--> 321\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m get_validated_options(opts, warn)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pymongo\\common.py:860\u001b[0m, in \u001b[0;36mget_validated_options\u001b[1;34m(options, warn)\u001b[0m\n\u001b[0;32m    858\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    859\u001b[0m     validator \u001b[38;5;241m=\u001b[39m URI_OPTIONS_VALIDATOR_MAP\u001b[38;5;241m.\u001b[39mget(normed_key, raise_config_error)\n\u001b[1;32m--> 860\u001b[0m     value \u001b[38;5;241m=\u001b[39m validator(opt, value)  \u001b[38;5;66;03m# noqa: PLW2901\u001b[39;00m\n\u001b[0;32m    861\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\u001b[38;5;167;01mValueError\u001b[39;00m, \u001b[38;5;167;01mTypeError\u001b[39;00m, ConfigurationError) \u001b[38;5;28;01mas\u001b[39;00m exc:\n\u001b[0;32m    862\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m warn:\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pymongo\\common.py:224\u001b[0m, in \u001b[0;36mvalidate_readable\u001b[1;34m(option, value)\u001b[0m\n\u001b[0;32m    221\u001b[0m \u001b[38;5;66;03m# First make sure its a string py3.3 open(True, 'r') succeeds\u001b[39;00m\n\u001b[0;32m    222\u001b[0m \u001b[38;5;66;03m# Used in ssl cert checking due to poor ssl module error reporting\u001b[39;00m\n\u001b[0;32m    223\u001b[0m value \u001b[38;5;241m=\u001b[39m validate_string(option, value)\n\u001b[1;32m--> 224\u001b[0m \u001b[38;5;28mopen\u001b[39m(value)\u001b[38;5;241m.\u001b[39mclose()\n\u001b[0;32m    225\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m value\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'isrgrootx1.pem'"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from pprint import pprint\n",
    "from pymongo.mongo_client import MongoClient\n",
    "\n",
    "mongodb_url_key = \"mongodb+srv://clarityai:N0lTZ4zedDxGbn9N@cluster0.qketnjc.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0&tlsCAFile=isrgrootx1.pem\"\n",
    "              \n",
    "\n",
    "# Create a new client and connect to the server\n",
    "client = MongoClient(mongodb_url_key)\n",
    "\n",
    "# Send a ping to confirm a successful connection\n",
    "try:\n",
    "    client.admin.command('ping')\n",
    "    print(\"Pinged your deployment. You successfully connected to MongoDB!\")\n",
    "except Exception as e:\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd21b7d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "print (client)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ead064c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "database = client['db-customer-churn']\n",
    "collection = database['telco-customer-churn']\n",
    "data = pd.read_csv(\"E:\\\\clarity.ai\\\\telco-customer-churn\\\\Telco-customer-churn.csv\")\n",
    "collection.insert_many(data.to_dict(orient='records'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f397e0bb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f96c7385",
   "metadata": {},
   "outputs": [],
   "source": [
    "collection = database['telco-customer-churn']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "175f538a",
   "metadata": {},
   "outputs": [],
   "source": [
    "collection"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2827a719",
   "metadata": {},
   "source": [
    "### Delete all documents from collection "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50ae5279",
   "metadata": {},
   "outputs": [],
   "source": [
    "collection.delete_many({})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fe4f3c99",
   "metadata": {},
   "outputs": [],
   "source": [
    "collection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c93b461c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
